low_level_platform/impl/src/lf_nrf52_support.c

```diff
@@ -258,23 +263,41 @@ int _lf_interruptable_sleep_until_locked(environment_t* env, instant_t wakeup_ti
   if (!_lf_async_event) {
     return 0;
   } else {
-    LF_PRINT_DEBUG("Sleep got interrupted...\n");
+    // LF_PRINT_DEBUG("Sleep got interrupted...\n");
     return -1;
   }
 }
 
+// Definition required by sd_nvic_critical_region_enter() and exit() below.
+nrf_nvic_state_t nrf_nvic_state = {0};
+
 /**
  * @brief Enter critical section. Let NRF Softdevice handle nesting
- * @return int
+ * @return 0
  */
-int lf_enable_interrupts_nested() { return sd_nvic_critical_region_enter(&_lf_nested_region); }
+int lf_enable_interrupts_nested() {
+  if (_lf_nested_count == 0)
+    return 1; // Error. Interrupts have not been disabled.
+  _lf_nested_count--;
+  return sd_nvic_critical_region_exit(0);
+  // FIXME: If softdevice is not enabled, do the following instead of above:
```

Comment:
Is there an issue for this? If not, we should create it, and reference it here.

Reply:
I have created an issue: https://github.com/lf-lang/reactor-c/issues/460